### PR TITLE
Add static loadJson method to Cesium3DTileset 

### PR DESCRIPTION
**Pull request by jason-crow**
This pull request adds a hook to Cesium's Cesium3DTileset's method for requesting the tileset json. The purpose of this is to make it easier to extend or decorate the method a tileset uses to fetch its json. For instance this facilitates fetching json's from remote servers that may require changing the base url or providing authentication headers. These are use cases that are essential to Bimium's use cases for creating tilesets with the Cesium library.

It doesn't seem necessary at the moment to also provide a hook for the TileContent request for the binary file, since this specific type of request is marked by Cesium.RequestType.TILES3D, it is already easily overridden without any specific hooks provided.

I provided 2 tests cases, the first just tests that Cesium3DTileset.loadJson by default loads json, the second test is more applicable to my use case for creating it. It tests to ensure the Cesium3DTileset.loadJson is used in the constructor for fetching the tileset json, as this is the assumption that makes decorating/extending it useful. To prove this it override the loadJson function and replaces the passed url which is invalid in the test with a valid one and tests whether the readyPromise resolves successfully to prove that overriding this method is useful for altering the method for fetching the tileset's json.

### CODE REVIEW

**Comment by pjcozzi** [comment]
Thanks @jason-crow!  @lilleyse will be the best person to review this.  It might take a bit as we are focused on prepping for SIGGRAPH next week.

File: Specs/Scene/Cesium3DTilesetSpec.js [hunk 1]
```diff
@@ -207,7 +207,49 @@ defineSuite([
...
+    it('static method loadJson is used in Cesium3DTileset constructor', function() {
+        var path = './Data/Cesium3DTiles/Tilesets/TilesetOfTilesets/tileset.json';
+
+        var originalLoadJson = Cesium3DTileset.loadJson;
+
+        // override loadJson and replace incorrect url with correct url
+        Cesium3DTileset.loadJson = function(tilesetUrl)
+            {
+            return originalLoadJson(path);
+            }
```
Comment (lilleyse) [inline Specs/Scene/Cesium3DTilesetSpec.js]: Fix formatting:

```
Cesium3DTileset.loadJson = function(tilesetUrl) {
    return originalLoadJson(path);
}
```

File: Source/Scene/Cesium3DTileset.js [hunk 2]
```diff
@@ -1049,6 +1049,14 @@ define([
     });
 
     /**
+     * Provides a hook to override the method used to request the tileset json
+     * useful when fetching tilesets from remote servers
+     */
```
Comment (lilleyse) [inline Source/Scene/Cesium3DTileset.js]: Add `@param` and `@returns` to the documentation.

**Comment by jason-crow** [comment]
Updated for the 2 changes you recommended. Should be ready to look at again.

File: Source/Scene/Cesium3DTileset.js [hunk 3]
```diff
@@ -1049,6 +1049,16 @@ define([
     });
 
     /**
+     * Provides a hook to override the method used to request the tileset json
+     * useful when fetching tilesets from remote servers
+     * @param {String} tilesetUrl, the url of the json file to be fetched
```
Comment (lilleyse) [inline Source/Scene/Cesium3DTileset.js]: Remove comma and capitalize `the`

File: Source/Scene/Cesium3DTileset.js [hunk 4]
```diff
@@ -1049,6 +1049,16 @@ define([
...
+     * @returns {Promise<Object>} A promise that resolves with the fetched json data
```
Comment (lilleyse) [inline Source/Scene/Cesium3DTileset.js]: I believe this needs to be `Promise.<Object>`

**Comment by jason-crow** [comment]
Done

**Comment by lilleyse** [comment]
Thanks @jason-crow

**Comment by pjcozzi** [comment]
Thanks again @jason-crow!

@lilleyse should you update CHANGES.md in master?

**Comment by lilleyse** [comment]
Yes, added here: https://github.com/AnalyticalGraphicsInc/cesium/commit/cce23920a4c366f4198584c171a51712d16a975c

**Pull request merged**